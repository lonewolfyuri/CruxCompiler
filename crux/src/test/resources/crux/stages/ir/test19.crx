// test array

array x:int[5];
  
func main() : void {
  var canary_begin:int;
  var canary_end:int;
  var outer:int;
  var inner:int;
  let canary_begin = 0;
  let canary_end = 0;

  let outer = 4;
  let inner = 2;
  while (outer >= 0) {
    while (inner >= 0) {
        let x[outer] = inner * 1000 + outer * 100 + 99;
      let inner = inner - 1;
    }
    let inner = 2;
    let outer = outer - 1;
  }

  let inner = 0;
  let outer = 0;
  while (outer < 5) {
    while (inner < 3) { 
       ::printInt(x[outer]);
       let inner = inner + 1;
     }
     ::println();
     let inner = 0;
     let outer = outer + 1;
  }
  ::printInt(canary_begin);
  ::printInt(canary_end);
}
	

