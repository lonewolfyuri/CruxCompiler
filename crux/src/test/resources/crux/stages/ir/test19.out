999999
199199199
299299299
399399399
499499499
00